cmd/gcp-controller-manager/node_syncer.go

```diff
@@ -154,12 +147,16 @@ func (ns *nodeSyncer) process(key string) error {
 		return fmt.Errorf("failed to get Pod %q: %v", key, err)
 	}
 	if !exists {
-		node, gsa, found := ns.nodes.remove(key)
-		if !found {
-			return nil
+		syncNodes := ns.nodes.remove(key)
+		klog.Infof("Pod %q being removed was found in %v nodes", key, len(syncNodes))
+		for node, gsa := range syncNodes {
+			klog.Infof("Pod %q and GSA %q are removed from Node %q", key, gsa, node)
```

Comment:
The log seems not accurate. Pod is removed, but the GSA may or may not be removed as there may be other pods that have GSA.

Reply:
You are right.  Removed gsa from the message to avoid confusion.

Reply:
Maybe we change it to something like 
      klog.Infof("Pod %q with GSA %q is removed from Node %q", key, gsa, node)
, so that we can grep GSA when debugging.